src/mlpack/methods/ann/activation_functions/lisht_function.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file lisht_function.hpp
+ * @author Kartik Dutt
+ *
+ * Definition and implementation of the LiSHT function as described by
+ * Swalpa K. Roy, Suvojit Manna, Shiv Ram Dubey and Bidyut B. Chaudhuri.
+ *
+ * For more information, see the following paper.
+ *
+ * @code
+ * @misc{
+ *   author = {Swalpa K. Roy, Suvojit Manna, Shiv R. Dubey and
+ *            Bidyut B. Chaudhuri},
+ *   title = {LiSHT: Non-Parametric Linearly Scaled Hyperbolic Tangent
+ *           Activation Function for Neural Networks},
+ *   year = {2019}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_LISHT_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_LISHT_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <algorithm>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The Lisht function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) = x * tanh(x)
+ * f'(x) = tanh(x) + x * (1 - tanh^{2}(x))
+ * @f}
+ */
+class LishtFunction
```

Comment:
Perhaps we should use the same spelling as used in the paper: LiSHT?

Reply:
Hi @zoq, I will make the changes.